cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/table/leveldb/LevelDBTableService.java

```diff
@@ -61,10 +61,8 @@
 
   private final ConcurrentMap<String, DB> tables = Maps.newConcurrentMap();
 
-  private static final LevelDBTableService SINGLETON = new LevelDBTableService();
-
   public static LevelDBTableService getInstance() {
-    return SINGLETON;
+    return new LevelDBTableService();
```

Comment:
This change is required. Previously `LevelDBTableService` was static, now we create it singleton per injector, so that each preview will have its own directories.